vpr/src/route/route_tree_timing.h

```diff
@@ -49,11 +49,14 @@ void print_route_tree_node(const t_rt_node* rt_root);
 void print_route_tree_inf(const t_rt_node* rt_root);
 void print_route_tree_congestion(const t_rt_node* rt_root);
 
-t_rt_node* traceback_to_route_tree(t_trace* head);
 t_rt_node* traceback_to_route_tree(ClusterNetId inet);
+t_rt_node* traceback_to_route_tree(ClusterNetId inet, std::vector<int>* non_config_node_set_usage);
+t_rt_node* traceback_to_route_tree(t_trace* head);
+t_rt_node* traceback_to_route_tree(t_trace* head, std::vector<int>* non_config_node_set_usage);
 t_trace* traceback_from_route_tree(ClusterNetId inet, const t_rt_node* root, int num_remaining_sinks);
 
 t_rt_node* prune_route_tree(t_rt_node* rt_root, CBRR& connections_inf);
+t_rt_node* prune_route_tree(t_rt_node* rt_root, CBRR& connections_inf, std::vector<int>* non_config_node_set_usage);
```

Comment:
We should have comments which clarify what the difference is between the versions which take `non_config_node_set_usage`, and those which don't.

For `traceback_to_route_tree()` I think it just records the node set usage (number of sinks connected via each set).

For `prune_troute_tree()` its less clear. Does the version without `non_config_node_set_usage` prune unused node sets or not? If they have different behaviour we should either give them different names or make very clear comments to that effect.

Reply:
I've updated the comments, let me know if you want further changes.